bundles/org.openhab.binding.wlanthermo/src/main/java/org/openhab/binding/wlanthermo/internal/api/esp32/WlanThermoEsp32CommandHandler.java

```diff
@@ -0,0 +1,280 @@
+                            return true;
+                        }
+                    case CHANNEL_COLOR:
+                        if (command instanceof HSBType) {
+                            channel.setColor(UtilEsp32.toHex((HSBType) command));
+                            return true;
+                        }
+                }
+            }
+        } else if (channelUID.getId().startsWith(CHANNEL_PITMASTER_PREFIX)) {
+            int channelId = Integer.parseInt(groupId.substring(CHANNEL_PITMASTER_PREFIX.length())) - 1;
+            if (data.getPitmaster() != null && data.getPitmaster().getPm() != null
+                    && data.getPitmaster().getPm().size() > channelId) {
+                Pm pm = data.getPitmaster().getPm().get(channelId);
+                switch (channelUID.getIdWithoutGroup()) {
+                    case CHANNEL_PITMASTER_CHANNEL_ID:
+                        pm.setChannel(((QuantityType<?>) command).intValue());
+                        return true;
+                    case CHANNEL_PITMASTER_PIDPROFILE:
+                        pm.setPid(((QuantityType<?>) command).intValue());
+                        return true;
+                    case CHANNEL_PITMASTER_SETPOINT:
+                        pm.setSet(((QuantityType<?>) command).toUnit(unit).doubleValue());
+                        return true;
+                    case CHANNEL_PITMASTER_STATE:
+                        String state = ((StringType) command).toString();
+                        if (state.equalsIgnoreCase("off") || state.equalsIgnoreCase("manual")
+                                || state.equalsIgnoreCase("auto")) {
+                            pm.setTyp(state);
+                            return true;
+                        }
+                }
+            }
+        }
+        return false;
+    }
+
+    public static String getTrigger(ChannelUID channelUID, @Nullable Data data)
+            throws WlanThermoUnknownChannelException, WlanThermoInputException {
+        if (channelUID.getGroupId() == null || data == null) {
+            throw new WlanThermoInputException();
+        }
+
+        String groupId = channelUID.getGroupId();
+        List<Channel> channelList = data.getChannel();
+
+        if (channelUID.getId().startsWith(CHANNEL_PREFIX)) {
+            int channelId = Integer.parseInt(groupId.substring(CHANNEL_PREFIX.length())) - 1;
+            if (channelList.size() > 0 && channelId < channelList.size()) {
+                Channel channel = channelList.get(channelId);
+                if (CHANNEL_ALARM_OPENHAB.equals(channelUID.getIdWithoutGroup())) {
+                    if (channel.getTemp() != 999) {
+                        if (channel.getTemp() > channel.getMax()) {
+                            return TRIGGER_ALARM_MAX;
+                        } else if (channel.getTemp() < channel.getMin()) {
+                            return TRIGGER_ALARM_MIN;
+                        }
+                    }
+                }
```

Comment:
I'd think that you would return something here just to avoid throwing a WlanThermoUnknownChannelException since it is clear at this point in the code that the channel is actually known.